### PR TITLE
Added notes on Filesystem Features

### PR DESCRIPTION
**Pull request by neilsjefferies**
First stab at #212

### CODE REVIEW

File: 0.1/implementation-notes/index.html [hunk 1]
```diff
@@ -165,7 +165,25 @@ <h2>Filesystem metadata</h2>
...
   </section>
-
+    
+  <section>
+    <h2>Filesystem Features</h2>
+    <p>
+      In order to be portable across as many filesystems as possible, OCFL makes use of a subset of filesystem features 
+      that are very broadly supported. It is therefore strongly advised to not use additional features in OCFL Storage
+      Roots since OCFL clients and other filesystem tools that need to operate between different filesystems may
+      exhibit unpredictable behaviour when feature sets do not match. In particular, using features such as hard and
```
Comment (ahankinson) [inline 0.1/implementation-notes/index.html]: 👍

File: 0.1/implementation-notes/index.html [hunk 2]
```diff
@@ -165,7 +165,25 @@ <h2>Filesystem metadata</h2>
...
+      that are very broadly supported. It is therefore strongly advised to not use additional features in OCFL Storage
+      Roots since OCFL clients and other filesystem tools that need to operate between different filesystems may
+      exhibit unpredictable behaviour when feature sets do not match. In particular, using features such as hard and 
+      soft (symbolic) links for deduplication can work at odds with OCFL's own mechanisms and should be avoided. 
+    </p>
+    <p>
+      Consideration should also be given to calculations of storage usage when migrating between filesystems. Many 
+      back-end filesystem features, which are essentially invisible to user-space code, can have a significant
+      impact on the actual consumption of storage space compared with the a simple sum of file sizes. Compression,
+      extents and block suballocation are examples of such features which, while providing benefits in terms of
```
Comment (ahankinson) [inline 0.1/implementation-notes/index.html]: probably 'sub-allocation' to be consistent (we use 'sub-directory' for example)

**Comment by ahankinson** [comment]
Looks good. I think that we'll need #227 as well for normative descriptions of certain filesystem features, such as path separators.

**Comment by ahankinson** [comment]
Ah, yes -- good catch. 

The decision on the editorial call was that `0.1/**/*` is effectively immutable, and all changes happen on the draft.

**Comment by neilsjefferies** [comment]
Reverting to apply to drafts

**Comment by zimeon** [comment]
Became #246

**Pull request closed without merging**